### PR TITLE
Add trust profile to UI.

### LINKED ISSUE #440: unstable crash in mdblist for sonarr
Hi,

I am testing the MDBList sync feature that was just released.

Radarr sync is not working as it is posting the list to the MDBList API but it is returning an error from MDBList. I have created a support request for MDBList to look into.

Sonarr does not work at all, it crashes. I am running in docker with unstable. It crashes with the below. When i turn debug on i can see it gathering the shows from sonarr but when it goes to process the shows it crashes the entire container. 

[INFO] 2023/02/16 16:09:25 !!> Stopping main Notifiarr loop. All timers and triggers are now disabled.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xbe3401]

goroutine 123 [running]:
github.com/Notifiarr/notifiarr/pkg/triggers/mdblist.(*cmd).getSonarrLibraries(0xc00079c488, {0x113e800, 0xc000280eb0}, 0xc001059860)
    github.com/Notifiarr/notifiarr/pkg/triggers/mdblist/mdblist.go:153 +0x621
github.com/Notifiarr/notifiarr/pkg/triggers/mdblist.(*cmd).sendMDBList(0xc00079c488, {0x113e800, 0xc000280eb0}, 0xc001059860)
    github.com/Notifiarr/notifiarr/pkg/triggers/mdblist/mdblist.go:84 +0xc8
github.com/Notifiarr/notifiarr/pkg/triggers/common.(*Config).runEventAction(0xc0007b5ea0?, {0x113e800?, 0xc000280eb0?}, 0xc001059860?, 0xc000dcbef0?)
    github.com/Notifiarr/notifiarr/pkg/triggers/common/timers.go:102 +0x14d
github.com/Notifiarr/notifiarr/pkg/triggers/common.(*Config).runTimerLoop(0xc0007b5ea0, {0x113e800, 0xc000280eb0}, {0xc0002cd200, 0x16, 0x0?}, {0xc0005f8000, 0x16, 0x20})
    github.com/Notifiarr/notifiarr/pkg/triggers/common/timers.go:86 +0xeb
created by github.com/Notifiarr/notifiarr/pkg/triggers/common.(*Config).Run
    github.com/Notifiarr/notifiarr/pkg/triggers/common/timers.go:50 +0x8ef

>  I just enabled mdblist sync.. then manully triggerd it via https://<my client instance>/ui/trigger/mdblist

Origin user: stevezau
Origin message: [Discord Server Link](https://discord.com/channels/764440599066574859/1075645553804841011/1075645553804841011)
Reporting user: TwitchCaptain
Reporting message: [Discord Server Link](https://discord.com/channels/764440599066574859/1075645553804841011/1075648182538739792)

### LINKED ISSUE #441: fix templating bug with tautulli URL.
I'm using Unraid Docker, when I'm in the local docker web interface and modify something and hit save I get this error: 
```
[ERROR] 2023/02/16 18:12:53 Sending HTTP Response: template: media/tautulli.html:77:300: executing "media/tautulli.html" at <.Config.Apps.Tautulli.URL>: reflect: indirection through nil pointer to embedded struct field Config

...

[ERROR] 2023/02/16 18:27:36 [gui 'xxx' requested] Copying Config (GUI request): encoding config into toml for copying (this is a bug!): template: config:353:25: executing "config" at <.Tautulli.URL>: reflect: indirection through nil pointer to embedded struct field Config
```

Origin user: AliG
Origin message: [Discord Server Link](https://discord.com/channels/764440599066574859/1075926081728675943/1075926081728675943)
Reporting user: TwitchCaptain
Reporting message: [Discord Server Link](https://discord.com/channels/764440599066574859/1075926081728675943/1075932934634229931)

### PR DESCRIPTION
**Pull request by davidnewhall**
This contribution adds the missing trust profile settings to the Web UI.

- Add `codespell` to lint and test.
- Adds `--reset` CLI arg to reset admin WebUI password.
- Closes #440.
- Closes #441.

### CODE REVIEW

File: pkg/bindata/templates/profile.html [hunk 1]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
                                         </tr>
                                     </thead>
                                     <tbody>
+                                        <tr>
+                                            <td>
+                                                <a onClick="dialog($(this), 'left')" class="help-icon far fa-question-circle"></a>
+                                                <span class="dialogTitle">Auth Type</span>
+                                                <div style="display:none;" class="dialogText">
+                                                    Three auth types are currently support.
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:26]:
```suggestion
                                                    Three auth types are currently supported.
```

File: pkg/bindata/templates/profile.html [hunk 2]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                    This configuration is not recommended.
+                                                    The No Password auth type allows any request from a known <code>upstream</code> IP.
+                                                    It also works with an auth proxy to set the username. If no username is in the provided header,
+                                                    then it defaults to admin.
+                                                </div>
+                                            </td>
+                                            <td>{{.Config.UIPassword.Type.String}}</td>
+                                            <td>
+                                                <select id="AuthType" name="AuthType" class="profile-parameter form-control input-sm" type="select">
+                                                    <option value="password"{{if eq .Config.UIPassword.Type 0}} selected{{end}}>Passowrd: Local Username</option>
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:52]:
```suggestion
                                                    <option value="password"{{if eq .Config.UIPassword.Type 0}} selected{{end}}>Password: Local Username</option>
```

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
Just some quick typos from a very quick skim

may be worth running codespell against the codebase again

File: examples/MANUAL.md [hunk 3]
```diff
@@ -61,6 +61,12 @@ OPTIONS
         This reads from stdin from dpkg, and sends it off to notifiarr.com.
         Will not work on any OS except Linux; only designed for Debian OSes.
 
+    --reset
+        This commands reads in the config file provided with --config,
```
Comment (bakerboy448) [inline examples/MANUAL.md:65]:
```suggestion
        This command reads in the config file provided with --config,
```

File: examples/MANUAL.md [hunk 4]
```diff
@@ -61,6 +61,12 @@ OPTIONS
...
+        changes the ui_password to a new random value and saves the file.
```
Comment (bakerboy448) [inline examples/MANUAL.md:66]: Does it output the password?

File: pkg/bindata/templates/profile.html [hunk 5]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
                                         </tr>
                                     </thead>
                                     <tbody>
+                                        <tr>
+                                            <td>
+                                                <a onClick="dialog($(this), 'left')" class="help-icon far fa-question-circle"></a>
+                                                <span class="dialogTitle">Auth Type</span>
+                                                <div style="display:none;" class="dialogText">
+                                                    Three auth types are currently supported.
+                                                    Auth Proxy and No Password are only available if the current request comes from a valid, known
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:27]:
```suggestion
                                                    Auth Proxy and No Password are only available if the current request comes from a valid, configured
```

File: pkg/bindata/templates/profile.html [hunk 6]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                    upstream IP address.
+                                                    <h4>Password: Local Username</h4>
+                                                    This is the default auth type.
+                                                    Select the Password option to log into this application with a configured username and password.
+                                                    Only 1 username and password is supported.
+                                                    When selecting this option, you must fill in the username and new password fields.
+                                                    If this is the current auth type you must also fill in the current password field.
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:34]: Fill in the current password field when....?

File: pkg/bindata/templates/profile.html [hunk 7]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                    <h4>Auth Proxy: Header</h4>
+                                                    Select this option to log into this application using an HTTP header that container a username.
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:36]:
```suggestion
                                                    Select this option to log into this application using an HTTP header that contains a username.
```

File: pkg/bindata/templates/profile.html [hunk 8]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                    This is commonly known as proxy authentication. We have
+                                                    <a href="https://notifiarr.wiki/en/Client/ReverseProxy">a wiki page</a>
+                                                    that provides instructions to configure Nginx for this feature.
+                                                    When selecting this option, you must also select a header for the auth proxy.
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:40]:
```suggestion
                                                    When selecting this option, in addition to setting trusted upstreams, you must also select a header for the auth proxy.
```

File: pkg/bindata/templates/profile.html [hunk 9]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                    If no headers are displayed, then no valid headers were provided by the upstream proxy.
+                                                    <h4>No Password</h4>
+                                                    This configuration is not recommended.
+                                                    The No Password auth type allows any request from a known <code>upstream</code> IP.
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:44]:
```suggestion
                                                    The No Password auth type allows any request from any configured <code>upstream</code> IP.
```

File: pkg/bindata/templates/profile.html [hunk 10]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                            <td class="mobile-hide">
+                                                {{range $i, $s := .Config.Allow.Input}}{{instance $i}}: <b>{{$s}}</b><br>{{end}}{{if not .Config.Allow.Input}}<i>no value, null</i>{{end}}
+                                            </td>
+                                            <td class="form-inline">
+                                                <div class="form-group" style="width:100%">
+                                                    {{- if (locked (printf "%s_UPTREAMS_0" .Flags.EnvPrefix))}}
+                                                    <div class="input-group" style="width:100%">
+                                                        <div style="width:30px; max-width:30px;" class="input-group-addon input-sm">
+                                                            <div style="display:none;" class="dialogText">
+                                                                An environment variable exists for this value. Your new value will write to the config file, but the application will not use it.
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:83]:
```suggestion
                                                                An environment variable exists for this value. Your new value will write to the config file, but the application will not use the config file value.
```

File: pkg/bindata/templates/profile.html [hunk 11]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                            </div>
+                                                            <a onClick="dialog($(this), 'right')" class="help-icon fas fa-outdent"></a>
+                                                            <span class="dialogTitle" style="display:none;">Env Variable: {{printf "%s_UPTREAMS_0" .Flags.EnvPrefix}}</span>
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:86]: UPTREAMS?

File: pkg/bindata/templates/profile.html [hunk 12]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
                                                                 <span class="dialogTitle" style="display:none;">Env Variable: {{printf "%s_UI_PASSWORD" .Flags.EnvPrefix}}</span>
                                                             </div>
-                                                            {{- end}}
-                                                            <input type="text" id="NewUsername" name="NewUsername" value="{{.Username}}" class="profile-parameter form-control input-sm" style="width: {{if (locked (printf "%s_UI_PASSWORD" .Flags.EnvPrefix))}}calc(100% - 15px){{else}}100%{{end}};">
+                                                            <input type="text" id="NewUsername" name="NewUsername" value="{{.Username}}" class="profile-parameter form-control input-sm" style="width:100%;">
+                                                        </div>
+                                                        {{ else }}
+                                                        <input type="text" id="NewUsername" name="NewUsername" value="{{.Username}}" class="profile-parameter form-control input-sm" style="width: 100%;">
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:142]: Confirm no spacing for newusername

File: pkg/bindata/templates/profile.html [hunk 13]
```diff
@@ -18,31 +18,143 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
...
+                                                <span class="dialogTitle">Auth Header</span>
+                                                <div style="display:none;" class="dialogText">
+                                                    This displays headers that are provided by the client or upstream that may be used for
+                                                    proxy authentication. This selection is only used if auth type is set to header.
+                                                    Choose the header that displays your proper username.
+                                                </div>
+                                            </td>
+                                            <td>{{if .Config.UIPassword.Webauth}}{{.Config.UIPassword.Header}}{{else}}none{{end}}</td>
+                                            <td>
+                                                <select id="AuthHeader" name="AuthHeader" class="profile-parameter form-control input-sm" type="select">
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:109]: AuthHeader?

File: pkg/bindata/templates/profile.html [hunk 14]
```diff
@@ -56,7 +168,7 @@ <h1><i class="fas fa-user-alt" style="float:left;font-size:40px"></i> User Profi
                                                                     <span class="dialogTitle" style="display:none;">Env Variable: {{printf "%s_UI_PASSWORD" .Flags.EnvPrefix}}</span>
                                                                 </div>
                                                             {{- end}}
-                                                            <input placeholder="16 character minimum" type="password" autocomplete="new-password" id="NewPassword" name="NewPassword" class="profile-parameter form-control input-sm" style="width: {{if (locked (printf "%s_UI_PASSWORD" .Flags.EnvPrefix))}}calc(100% - 15px){{else}}100%{{end}};">
+                                                            <input placeholder="9 character minimum" type="password" autocomplete="new-password" id="NewPassword" name="NewPassword" class="profile-parameter form-control input-sm" style="width: {{if (locked (printf "%s_UI_PASSWORD" .Flags.EnvPrefix))}}calc(100% - 15px){{else}}100%{{end}};">
```
Comment (bakerboy448) [inline pkg/bindata/templates/profile.html:171]: NewPassword, no space?

File: pkg/client/handlers_gui.go [hunk 15]
```diff
@@ -377,21 +413,17 @@ func (c *Client) handleProfilePost(response http.ResponseWriter, request *http.R
...
+		http.Error(response, "Saving Trust Profile: "+err.Error(), http.StatusInternalServerError)
 
 		return
 	}
 
-	c.Printf("[gui '%s' requested] Updated primary username and password, new username: %s", currUser, username)
+	c.Printf("[gui '%s' requested] Updated Trust Profile settings, username: %s", currUser, username)
 	c.setSession(username, response)
-	http.Error(response, "New username and/or password saved.", http.StatusOK)
+	http.Error(response, "Trust profile saved.", http.StatusOK)
```
Comment (bakerboy448) [inline pkg/client/handlers_gui.go:425]:
```suggestion
	http.Error(response, "Trust Profile saved.", http.StatusOK)
```

**Review by bakerboy448 — CHANGES_REQUESTED** [review]

File: examples/MANUAL.md (hunk 4 above)
Comment (davidnewhall) [reply examples/MANUAL.md:66]: Si.

File: pkg/bindata/templates/profile.html (hunk 6 above)
Comment (davidnewhall) [reply pkg/bindata/templates/profile.html:34]: When it tells you your password is wrong. haha

File: pkg/bindata/templates/profile.html (hunk 8 above)
Comment (davidnewhall) [reply pkg/bindata/templates/profile.html:40]: It's clear you haven't tried this yet. You can't select a header until you set trusted upstreams.

File: pkg/bindata/templates/profile.html (hunk 10 above)
Comment (davidnewhall) [reply pkg/bindata/templates/profile.html:83]: If you want this changed, open a PR and change every instance.

**Review by bakerboy448 — APPROVED** [review]

**Pull request merged**